lib/cacheWrapper.js

```diff
@@ -112,32 +155,60 @@ cacheWrapper._queueRequest = function( deferred, options ) {
 * @param {Object} options - Including segment, key and value.
 * @returns {Object} promise
 */
-cacheWrapper._addCallToQueue = function( options ) {
+cacheWrapper._addCallToQueue = function(options) {
+  if (!cacheClient) {
+    return Q.reject(new Error('Cache not initialised'));
+  }
   var deferred = Q.defer();
-  if ( !cacheClient ) {
-    deferred.reject( new Error( 'Cache not initialised' ) );
+  if (!cacheClient.isReady()) {
+    // cache isnt ready, queue it up
+    cacheWrapper._queueRequest(deferred, options);
   } else {
-    if ( !cacheClient.isReady() ) {
-      // cache isnt ready, queue it up
-      cacheWrapper._queueRequest( deferred, options );
+    // execute the requested method straight away
+    if (options.method === 'stash') {
+      cacheWrapper._stash(deferred, options);
     } else {
-      // execute the requested method straight away
-      if ( options.method === 'stash' ) {
-        cacheWrapper._stash( deferred, options );
-      } else {
-        cacheWrapper._retrieve( deferred, options );
-      }
+      cacheWrapper._retrieve(deferred, options);
     }
   }
   return deferred.promise;
 };
 
-cacheWrapper.set = function( options ) {
+/**
+* Allows you to add a `key/value` pair to the cache
+* @param {Object} options - Including segment, key and value.
+* @returns {Object} promise
+*/
+cacheWrapper.set = function(options) {
   options.method = 'stash';
-  return cacheWrapper._addCallToQueue( options );
+  return cacheWrapper._addCallToQueue(options);
 };
 
-cacheWrapper.get = function( options ) {
+/**
+* Allows you to retrieve a value from the cache for the passed `key`
+* @param {Object} options - Including segment and key.
+* @returns {Object} promise
+*/
+cacheWrapper.get = function(options) {
   options.method = 'retrieve';
-  return cacheWrapper._addCallToQueue( options );
+  return cacheWrapper._addCallToQueue(options);
+};
+
+/**
+* Drops records from the cache with the given prefix
+* @param {Object} options - Including segment and prefix.
+* @returns {Object} promise
+*/
+cacheWrapper.delete = function(options) {
```

Comment:
another way to approach this which would be simpler/less code would be after the `SCAN` call, instead of calling `delete()` which is a new function for this feature, scrap that and call `set` with a `0` TTL. Just a thought.